File: ui/src/app/model/CategoryAdminDashboardVO.ts
import { IncidentModuleVO } from "./IncidentModuleVO";
import { IncidentPriorityVO } from "./IncidentPriorityVO";
import { IncidentStatusVO } from "./IncidentStatusVO";

export class CategoryAdminDashboardVO {

    priorityWiseIncidents: IncidentPriorityVO[];
    moduleWiseIncidents: IncidentModuleVO[];
    statusWiseIncidents: IncidentStatusVO[];
    aging5: number;
    aging10: number;
    aging20: number;
    aging30: number;
    aging60: number;
    constructor(data: any) {
        if (data == null) {
            this.moduleWiseIncidents = [];
            this.priorityWiseIncidents = [];
            this.statusWiseIncidents = [];
            this.aging5 = -1;
            this.aging10 = -1;
            this.aging20 = -1;
            this.aging30 = -1;
            this.aging60 = -1;
        } else {
            this.moduleWiseIncidents = data.moduleWiseIncidents;
            this.priorityWiseIncidents = data.priorityWiseIncidents;
            this.statusWiseIncidents = data.statusWiseIncidents;
            this.aging5 = data.aging5;
            this.aging10 = data.aging10;
            this.aging20 = data.aging20;
            this.aging30 = data.aging30;
            this.aging60 = data.aging60;
        }

    }

}

File: domain/src/main/java/com/centram/common/vo/CategoryAdminDashboardVO.java
package com.centram.common.vo;


import lombok.*;

import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class CategoryAdminDashboardVO {
    private Integer aging5;
    private Integer aging10;
    private Integer aging20;
    private Integer aging30;
    private Integer aging60;
    private Set<IncidentPriorityVO> priorityWiseIncidents;
    private Set<IncidentModuleVO> moduleWiseIncidents;
    private Set<IncidentStatusVO> statusWiseIncidents;
    private Set<IncidentPriorityVO> priorityWiseAssetIncidents;
    private Set<IncidentModuleVO> moduleWiseAssetIncidents;
    private Set<IncidentStatusVO> statusWiseAssetIncidents;

    public CategoryAdminDashboardVO(Integer aging5, Integer aging10, Integer aging20, Integer aging30, Integer aging60) {
        this.aging5 = aging5;
        this.aging10 = aging10;
        this.aging20 = aging20;
        this.aging30 = aging30;
        this.aging60 = aging60;
    }
}